src/modules/job-info/job-info.c

```diff
@@ -307,7 +310,40 @@ static void lookup_cb (flux_t *h, flux_msg_handler_t *mh,
         goto error;
     }
 
-    if (!(l = lookup_ctx_create (ctx, msg, id, flags)))
+    if (!(l = lookup_ctx_create (ctx, msg, id, flags, false)))
+        goto error;
+
+    if (lookup_key (l) < 0)
+        goto error;
+
+    if (zlist_append (ctx->lookups, l) < 0) {
+        flux_log_error (h, "%s: zlist_append", __FUNCTION__);
+        goto error;
+    }
+    zlist_freefn (ctx->lookups, l, lookup_ctx_destroy, true);
+    l = NULL;
```

Comment:
unnecessary?

Reply:
nah, it is a sort of personal style to say "this is now owned by someone else", but given we return right after, no need to set it.